apstra/api_systems_configuration.go

```diff
@@ -0,0 +1,92 @@
+package apstra
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"time"
+)
+
+const (
+	apiUrlSystemsConfiguration = apiUrlSystemsById + apiUrlPathDelim + "configuration"
+)
+
+var _ json.Unmarshaler = new(SystemConfig)
+
+type SystemConfig struct {
+	SystemId ObjectId
+	//DeployState                   string
+	//ConfigurationServiceState     string
+	LastBootTime                  time.Time
+	Deviated                      bool
+	ErrorMessage                  *string
+	ContiguousFailures            uint32
+	UserGoldenConfigUpdateVersion uint32
+	UserFullConfigDeployVersion   uint32
+	AosConfigVersion              uint32
+	ExpectedConfig                string
+	ActualConfig                  string
+}
+
+func (o *SystemConfig) UnmarshalJSON(bytes []byte) error {
+	var raw struct {
+		SystemId ObjectId `json:"system_id"`
+		//DeployState                   string   `json:"deploy_state"`
+		//ConfigurationServiceState     string   `json:"configuration_service_state"`
+		LastBootTime                  float64 `json:"last_boot_time"`
+		Deviated                      bool    `json:"deviated"`
+		ErrorMessage                  string  `json:"error_message,omitempty"`
+		ContiguousFailures            uint32  `json:"contiguous_failures"`
+		UserGoldenConfigUpdateVersion uint32  `json:"user_golden_config_update_version"`
+		UserFullConfigDeployVersion   uint32  `json:"user_full_config_deploy_version"`
+		AosConfigVersion              uint32  `json:"aos_config_version"`
+		Expected                      struct {
+			Config string `json:"config"`
+		} `json:"expected"`
+		Actual struct {
+			Config string `json:"config"`
+		} `json:"actual"`
+	}
+
+	err := json.Unmarshal(bytes, &raw)
+	if err != nil {
+		return err
+	}
+
+	sec, nsec := math.Modf(raw.LastBootTime)
```

Comment:
Is the primary reason for this unmarshal to convert the last boot time into unix time?

Reply:
This object implements `json.Unmarshaler` interface, so there's no need for a `rawThing` nor a `polish()` method.

It's all self-contained in the public object (something I wish I'd known about a couple of years ago).